### ISSUE TITLE
Terraform volume extend for centos 7

### ISSUE THREAD

**Issue by sujith-siva** [hashicorp/terraform#20569]
Hi there,

I have a requests received for extending a volume from 200GB to 1TB for an existing postgreSQL ec2 instance. I have changed the volume from my existing tf file from 200 to 1000 and it didn't do anything.
Now when I read regarding the issues I have found out that when I next time apply the terraform plan it will destroy the existing instance and create a new one which is a disaster for me as its a production server and I don't want to lose any of the existing files/folders inside that drive.

So I need a solution first not to destroy my existing instance with the same files in the drive then a feasible solution for extending the drive to 1000 GB without affecting the existing drive files.

df -h shows
/dev/mapper/data  197G   29G  158G  16% /var/lib/pgsql

Thanks
Sujith Siva

**Comment by sujith-siva** [comment]
https://www.terraform.io/docs/commands/taint.html

**Comment by mildwonkey** [comment]
Hi @sujith-siva,
This sounds like a task that requires human intervention and is best handled outside of terraform.  You can manually detach the volume and use the [`terraform state rm`](https://www.terraform.io/docs/commands/state/rm.html) command to keep terraform from destroying it.
The next steps will depend on your needs. Without knowing all of the details, I can only make general suggestions: You could run `terraform apply` to create and attach the new volume, then manually run whatever commands needed to copy the files to that volume.

**Comment by sujith-siva** [comment]
thanks for the help.
If I attach a new volume by running terraform apply again doesn't break my existing drive which contains files/folders?

**Comment by mildwonkey** [comment]
Not if you run `terraform state rm [volume address]` first. That tells terraform to 'forget' that the original volume exists. This does mean that you will need to manually delete the volume when you are finished with it.

**Comment by sujith-siva** [comment]
Hi,

if I run terraform state rm [volume address] which tells terraform to "forget about the volume with that address". So after I removed that entry from state file and apply a new change by adding a new drive to the instance does it keep the existing volume with 200GB as I had before without any data loss.

I don't want to remove the existing volume in stead I need to extend the same volume which is using LVM from 200 GB to 1000 GB.

Thanks again for the reply.

**Comment by mildwonkey** [comment]
Ah, ok. In that case I believe you can extend the volume manually and update your terraform configuration to match the new size, after which `terraform plan` should show no changes. 

This does depend heavily on your exact configuration and provider, so you will want to test this in a dev setup first.

**Comment by sujith-siva** [comment]
I have found a fix now.
I have changed my original instance.tf file with the data drive as 200.
After that I have created a new .tf file for a new ebs_block device and then created another code in the same tf file for attaching this to the existing instance so now it doesn't showing as destroying the original drive in stead keeping the original drive plus adding another drive of 1000GB to the same instance.

Now I think the existing drive hasn't broken and seems to have another bigger data drive I have added to the same instance so that I can use lvextend to extend the existing volume group after I have created a new physical volume with the new added drive(1000 GB).

thanks for your help.
